src/main/java/io/netty/incubator/codec/http3/Http3PushStreamServerInitializer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.http3;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.ChannelPromise;
+import io.netty.incubator.codec.quic.QuicStreamChannel;
+
+import static io.netty.incubator.codec.http3.Http3CodecUtils.writeVariableLengthInteger;
+import static io.netty.incubator.codec.http3.Http3RequestStreamDecodeState.NO_DECODE_STATE;
+import static io.netty.util.internal.ObjectUtil.checkPositiveOrZero;
+
+/**
+ * Abstract base class that users can extend to init HTTP/3 push-streams for servers. This initializer
+ * will automatically add HTTP/3 codecs etc to the {@link ChannelPipeline} as well.
+ */
+public abstract class Http3PushStreamServerInitializer extends Http3PushStreamInitializer {
+
+    private final long pushId;
+
+    protected Http3PushStreamServerInitializer(long pushId) {
+        this.pushId = checkPositiveOrZero(pushId, "pushId");
+    }
+
+    @Override
+    protected final void initChannel(QuicStreamChannel ch) {
+        // We need to write stream type into the stream before doing anything else.
```

Comment:
should we validate that this is really server side ?

Reply:
Is there a way to check this using the `QuicStreamChannel`? I am not sure if it is worth.

Reply:
Sure you can just check if streamId is odd or not.. if odd it is server side. 

Like:

```
boolean isServer = ch.streamId() % 2 != 0;
```

Reply:
do the same above to verify if its client side